### PR TITLE
[Bugfix] sort columns not working in submissions-dashboards

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

The sorting buttons in the assessment-dashboards did not work. In this PR the problem is solved.

image

Also a bug where no locked submissions could be cancelled is fixed here. There was a mistake in a check.
### Description

I fixed the column sorting for the modeling/file-upload/text/programming-assessment-dashboards.

### Steps for Testing

1. In a course under exercises check under "submissions" if all sort buttons work, for each exercise type.
2. Same for exam -> exercise-groups -> "Submissions"

### CODE REVIEW

**Review by airelawaleria — COMMENTED** [review]
Sorting didn't work for Points at all, for Bonus Points, Presentation and Publish Build Plan only once (first click). Also, is the Participation Mode sorting working/necessary :) ?

Everything else worked :)

**Review by JohannesWeiss — COMMENTED** [review]
Works mostly as expected, but I encountered the same incorrect behaviours as @airelawaleria with (bonus) points.

**Comment by SimonEntholzer** [comment]
> Sorting didn't work for Points at all, for Bonus Points, Presentation and Publish Build Plan only once (first click). Also, is the Participation Mode sorting working/necessary :) ?
> 
> Everything else worked :)

Ah, maybe my screenshots were a bit misleading :D I did not change anything in the course overview, neither in the exercise group overview. The only changes are in the assessment dashboards which you access over the "submission" button :)

**Review by airelawaleria — COMMENTED** [review]
is it intended that "submissions" from unsubmitted exams (that are assessed automatically) do not participate in sorting?
PS. in the screenshot sorted by Latest Result
image

**Comment by SimonEntholzer** [comment]
> is it intended that "submissions" from unsubmitted exams (that are assessed automatically) do not participate in sorting?
> PS. in the screenshot sorted by Latest Result
> 

I guess it is intended, but I'm not sure. The problem is that you can't compare undefined values with other elements so it makes sense to just leave them where they are.

**Review by airelawaleria — APPROVED** [review]
everything worked now. 

There is no sorting possible for modelling submissions, but it's the same on develop, thus approve :)

**Review by JohannesWeiss — APPROVED** [review]
Tested on ts1, sorting seems fine now. The issue with upload exercises is fixed as well.

File: src/main/webapp/app/shared/service/sort.service.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export class SortService {
             const valueA = this.customGet(a, key, undefined);
             const valueB = this.customGet(b, key, undefined);
 
-            if (!valueA || !valueB) {
+            if ((!valueA && valueA !== 0) || (!valueB && valueB !== 0)) {
```
Comment (krusche) [inline src/main/webapp/app/shared/service/sort.service.ts:15]: I fixed this differently in another PR

Comment (SimonEntholzer) [reply src/main/webapp/app/shared/service/sort.service.ts:15]: ah yea, the `== undefined` is definitely cleaner

**Review by airelawaleria — APPROVED** [review]

**Review by simonzilker — APPROVED** [review]

**Review by marvinchimeka — APPROVED** [review]

**Review by EneaGore — APPROVED** [review]

**Review by JohannesWeiss — APPROVED** [review]

**Review by nicolasruscher — APPROVED** [review]
Looks good!

**Review by FuchsDominik — APPROVED** [review]
Couldn't find any sorting which does not work. Code also looks fine :)

**Pull request merged**